### ISSUE TITLE
Segfault in Jhead at exif.c:418

### ISSUE THREAD

**Issue by Ngiong** [Matthias-Wandel/jhead#2]
Hello,
We are currently working on a new fuzz testing feature, and we found a crash in jhead.

**Step to Reproduce**
We used `CFLAGS="-g -O0" make` to compile the jhead, and run it with:
```
./jhead -norot -exonly <attached_file> -purejpg -c -autorot
```
Attachment: [poc_0001.txt](https://github.com/Matthias-Wandel/jhead/files/5420120/poc_0001.txt)

**Environment**
 - OS: Ubuntu 18.04.4 LTS
 - Compiler: gcc 7.5.0
 - jhead version: 3.04 (https://www.sentex.ca/~mwandel/jhead/jhead-3.04.tar.gz)

**Additional context**
Here's the stack trace: [stack_trace_0001.txt](https://github.com/Matthias-Wandel/jhead/files/5420353/stack_trace_0001.txt)

```
Program received signal SIGSEGV, Segmentation fault.
ConvertAnyFormat (ValuePtr=0x1617508, Format=<optimized out>) at exif.c:418
418	exif.c: No such file or directory.
#0  ConvertAnyFormat (ValuePtr=0x1617508, Format=<optimized out>) at exif.c:418
#1  0x0000000000408c2c in ProcessGpsInfo (DirStart=<optimized out>, OffsetBase=<optimized out>, ExifLength=<optimized out>) at gpsinfo.c:165
#2  0x000000000040717a in ProcessExifDir (DirStart=<optimized out>, OffsetBase=<optimized out>, ExifLength=<optimized out>, NestingLevel=<optimized out>) at exif.c:866
#3  0x0000000000406375 in process_EXIF (ExifSection=0x617500 "\001RExif", length=338) at exif.c:1041
#4  0x000000000040401d in ReadJpegSections (infile=0x6162c0, ReadMode=<optimized out>) at jpgfile.c:287
#5  0x00000000004046e8 in ReadJpegFile (FileName=0x7fffffffe244 "input", ReadMode=READ_METADATA) at jpgfile.c:379
#6  0x000000000040240c in ProcessFile (FileName=<optimized out>) at jhead.c:905
#7  main (argc=<optimized out>, argv=<optimized out>) at jhead.c:1756
```

Thank you.

**Comment by Matthias-Wandel** [comment]
Fixed by change 56c108f1a0653b3a3c98960f66c2de5a9a67e5de on 
https://github.com/Matthias-Wandel/jhead

------ Original Message ------

**Comment by Matthias-Wandel** [comment]
Fixed by 56c108f1a0653b3a3c98960f66c2de5a9a67e5de
